### PR TITLE
XUnit and NUnit integration with examples

### PR DESCRIPTION
**Pull request by alkondrashov**
- Added example for NUnit in C# and for XUnit in F#

- API to be used for tests F#:
`HttpScenario.buildScenario() |> Scenario.applyAssertions(assertions)`

- For C#:
`ScenarioRunner.ApplyAssertions(scenario, assertions);`

- The same API is for both NUnit and XUnit

- Added `ScenarioRunner` static class to `CSharp` api to shield Infra/ScenarioRunner.fs from clients

- in `CSharp` api renamed: `Assert` to `AssertFor` because there was a conflicts in names with `XUnit.Assert` (for `FSharp` api Assert is good). Example of updated API:

```
AssertFor.Scenario(stats => stats.OkCount >= 95),
AssertFor.TestFlow("Flow name 23", stats => stats.FailCount < 10),
AssertFor.TestFlow("test flow1", stats => stats.FailCount < 10),
AssertFor.Step("Flow name 1", "step name 10", stats => stats.FailCount == 95),
```

- Had to copy `HttpScenario.fs` across test projects for now; later to be replaced with NBomber.Http

Related to issue #47

### CODE REVIEW

**Comment by alkondrashov** [comment]
Alternatives for API:
```
HttpScenario.buildScenario() |> Scenario.apply(assertions)
ScenarioRunner.Apply(scenario, assertions);
```

OR

```
HttpScenario.buildScenario() |> Scenario.assert'(assertions)
ScenarioRunner.Assert(scenario, assertions);
```

?

File: src/NBomber/Infra/Assertions.fs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        let ty = Type.GetType("Xunit.Assert, xunit.assert") //xunit v2
+        if ty <> null then Xunit ty else
+
+        let ty = Type.GetType("NUnit.Framework.Assert, nunit.framework")
+        if ty <> null then Nunit ty else
+
+        Generic
+
+    match framework with
+        | Xunit ty ->
```
Comment (AntyaDev) [inline src/NBomber/Infra/Assertions.fs]: match framework with
| Xunit ty ->

File: examples/NBomber.Examples.CSharp/Program.cs [hunk 2]
```diff
@@ -15,19 +15,19 @@ static void Main(string[] args)
             //var scenario = HttpScenario.BuildScenario();  
             var scenario = SimplePushScenario.BuildScenario();
 
-            var asserts = new [] {
```
Comment (AntyaDev) [inline examples/NBomber.Examples.CSharp/Program.cs]: Asser.ForScenario looks better than AssertFor
+ I  assume that letter you could add something else for just plain Assert, like Asser.IsTrue, for such case AssertFor will be strange

Comment (alkondrashov) [reply examples/NBomber.Examples.CSharp/Program.cs]: Maybe Assertion.ForScenario ?

**Pull request merged**